ntbs-service/DataMigration/NotificationImportService.cs

```diff
@@ -306,38 +307,54 @@ private void LookupAndAssignPostcode(List<Notification> notifications)
 
         private async Task<IEnumerable<ValidationResult>> GetValidationErrors(Notification notification)
         {
-            var validationsResults = new List<ValidationResult>();
-
+            var singletonModels = new List<ModelBase>
+            {
+                notification.PatientDetails,
+                notification.ClinicalDetails,
+                notification.TravelDetails,
+                notification.VisitorDetails,
+                notification.ComorbidityDetails,
+                notification.ImmunosuppressionDetails,
+                notification.SocialRiskFactors,
+                notification.HospitalDetails,
+                notification.ContactTracing,
+                notification.PatientTBHistory,
+                notification.TestData,
+                notification.MBovisDetails
+            };
+            var modelCollections = new List<ICollection<ModelBase>>
+            {
+                notification.TestData.ManualTestResults as ICollection<ModelBase>,
+                notification.SocialContextAddresses as ICollection<ModelBase>,
+                notification.SocialContextVenues as ICollection<ModelBase>,
+                notification.TreatmentEvents as ICollection<ModelBase>,
+                notification.MBovisDetails.MBovisAnimalExposures as ICollection<ModelBase>,
+                notification.MBovisDetails.MBovisExposureToKnownCases as ICollection<ModelBase>,
+                notification.MBovisDetails.MBovisOccupationExposures as ICollection<ModelBase>,
+                notification.MBovisDetails.MBovisUnpasteurisedMilkConsumptions as ICollection<ModelBase>,
+            };
+            
+            // Set correct validation context everywhere
             NotificationHelper.SetShouldValidateFull(notification);
+            void SetValidationContext(ModelBase model) => model.SetValidationContext(notification);
```

Comment:
hah, didn't know you can do this within the method.

Reply:
yeah, the feature is "local function" if you wanted to look it up further :)